lnst/RecipeCommon/Perf/Measurements/Results/AggregatedRDMABandwidthMeasurementResults.py

```diff
@@ -0,0 +1,32 @@
+from typing import Optional
+from lnst.RecipeCommon.Perf.Measurements.Results.BaseMeasurementResults import BaseMeasurementResults
+from lnst.RecipeCommon.Perf.Measurements.Results.RDMABandwidthMeasurementResults import RDMABandwidthMeasurementResults
+from lnst.RecipeCommon.Perf.Results import SequentialPerfResult
+
+
+class AggregatedRDMABandwidthMeasurementResults(BaseMeasurementResults):
+    def __init__(self, results: Optional[list[RDMABandwidthMeasurementResults]] = None):
+        self._individual_results = results or []
+
+    @property
+    def flow(self):
+        return self._individual_results[0].flow
+
+    @property
+    def bandwidth(self) -> SequentialPerfResult:
+        return SequentialPerfResult([result.bandwidth for result in self.individual_results])
+
+    @property
+    def individual_results(self) -> list[RDMABandwidthMeasurementResults]:
+        return self._individual_results
+
+    def add_results(self, result: RDMABandwidthMeasurementResults) -> None:
+        self._individual_results.append(result)
+
+    def describe(self) -> str:
```

Comment:
Yee there's an issue with typing (and logic), because the classes are not the same. When we force inheritance like this, we force the non aggregated result to be possibly not initialized from the constructor. I think it would clean up the code if we used two separate classes AND a protocol connecting them.

```py
from dataclasses import dataclass, field
from typing import Protocol


class Result(Protocol):
    @property
    def bandwidth(self) -> float:
        ...

@dataclass
class SingleResult(Result):  # inheritance optional
    bandwidth: float
    

class AggregatedResult(Result):  # inheritance optional
    results: list[Result] = field(default_factory=list)
    
    @property
    def bandwidth(self) -> float:
        return sum(result.bandwidth for result in self.results) / len(self.results)
```
This should for sure be its own MR though.

For now, should I make `AggregatedRDMABandwidthMeasurementResults` inehrit from `RDMABandwidthMeasurementResults` and make `RDMABandwidthMeasurementResults` possible to be instantiated without any actual bandwidth, possibly making it raise an exception if any other methods were called on it, making it the same as all other results/aggregated results in LNST?

Reply:
> For now, should I make AggregatedRDMABandwidthMeasurementResults inehrit from RDMABandwidthMeasurementResults

yeah, let's keep it consistent for now.

i agree with a separate MR for a more complex refactor of the `MeasurementResults` classes, i also mentioned some "proposal" for this on our internal slack, but let's have that as a separate discussion

Reply:
fixed.